### PR TITLE
chase-symlinks: Add chase_and_open_parent() + at() variant

### PR DESCRIPTION
**Pull request by daandemeyer**
Split out of #26707

### CODE REVIEW

**Review by bluca — APPROVED** [review]

**Comment by DaanDeMeyer** [comment]
No point in waiting for integration tests since this isn't actually used anywhere yet.

File: src/basic/chase.c [hunk 1]
```diff
@@ -784,6 +784,18 @@ int chase_and_unlink(
         return 0;
 }
 
+int chase_and_pin(const char *path, const char *root, ChaseFlags chase_flags, char **ret_filename) {
+        int pfd, r;
+
+        assert(!(chase_flags & (CHASE_NONEXISTENT|CHASE_STEP|CHASE_PARENT|CHASE_EXTRACT_FILENAME)));
```
Comment (yuwata) [inline src/basic/chase.c]: In the below, CHASE_PARENT and CHASE_EXTRACT_FILENAME are added. Hence, it is not necessary to set it by the caller, but I do not think it is not necessary to trigger the assertion...
So, please drop the two flags from the assertion.

File: src/basic/chase.c [hunk 2]
```diff
@@ -1003,3 +1015,20 @@ int chase_and_unlinkat(
...
+
+int chase_and_pinat(
+                int dir_fd,
+                const char *path,
+                ChaseFlags chase_flags,
+                char **ret_filename) {
+
+        int pfd, r;
+
+        assert(!(chase_flags & (CHASE_NONEXISTENT|CHASE_STEP|CHASE_PARENT|CHASE_EXTRACT_FILENAME)));
```
Comment (yuwata) [inline src/basic/chase.c]: Also here.

File: src/basic/chase.c [hunk 3]
```diff
@@ -784,6 +784,18 @@ int chase_and_unlink(
         return 0;
 }
 
+int chase_and_pin(const char *path, const char *root, ChaseFlags chase_flags, char **ret_filename) {
```
Comment (yuwata) [inline src/basic/chase.c]: Hmm... Maybe, `chase_and_open_parent()` is better??

**Review by yuwata — CHANGES_REQUESTED** [review]

**Comment by DaanDeMeyer** [comment]
Addressed comments, changes are trivial, so setting the green label.

**Pull request merged**